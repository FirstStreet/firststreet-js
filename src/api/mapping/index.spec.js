const sut = require('./index');

describe('model mapping', () => {
  it('can return mapping', () => {
    const result = sut('location', 'detail');
    expect(result).not.toBeUndefined();
    expect(result).toMatchSnapshot();
    expect(result.model.property.name).toEqual('LocationPropertyDetail');
  });

  it('should throw error when unknown service name', () => {
    expect(() => sut('unknown', 'detail')).toThrow('Invalid service unknown. Allowed: location, probability, historic, economic');
  });

  it('should throw error when unknown detail level', () => {
    expect(() => sut('location', 'unknown')).toThrow('Invalid detail level unknown for service location. Allowed: detail, summary');
  });
});
