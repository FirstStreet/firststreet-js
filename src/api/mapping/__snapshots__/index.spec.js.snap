// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model mapping can return mapping 1`] = `
Object {
  "endpointPrefix": "/location/detail",
  "model": Object {
    "cd": [Function],
    "city": [Function],
    "county": [Function],
    "neighborhood": [Function],
    "property": [Function],
    "state": [Function],
    "tract": [Function],
    "zcta": [Function],
  },
  "needsLocation": true,
}
`;
